src/geop/services/src/main/scala/com/azavea/mmw/GeopServiceActor.scala

```diff
@@ -0,0 +1,69 @@
+package com.azavea.mmw
+
+import akka.actor.Actor
+import spray.routing._
+import spray.http._
+
+object UnattractiveButTemporary {
+  import geotrellis.raster.io.geotiff.reader._
+  import geotrellis.raster._
+  import geotrellis.raster.reproject._
+  import geotrellis.vector._
+  import geotrellis.vector.reproject._
+  import geotrellis.vector.io.json._
+  import geotrellis.spark.utils.SparkUtils
+  import geotrellis.spark._
+  import geotrellis.spark.tiling._
+  import geotrellis.raster.mosaic._
+  import geotrellis.proj4._
+
+  import org.apache.spark._
+  import org.apache.spark.{SparkConf, SparkContext}
+
+  import spray.json._
+  import spray.json.DefaultJsonProtocol._
+  import spray.httpx.unmarshalling._
+
+  import DataFiles._
+  import NonHistogram._
+
+  implicit val sc: SparkContext = SparkUtils.createSparkContext("Geoprocessing", new SparkConf())
+  val nlcd = DataFiles.delawareNLCD
+  val soil = DataFiles.delawareSoil
+
+  def jsonHistogram(str: String) : String = {
+    val polygon = str.parseJson.convertTo[Polygon]
+    val histogram: List[((Int, Int), Int)] = NonHistogram.gtHistogram(nlcd, soil, polygon).toList
+    histogram.toJson.prettyPrint
+  }
+}
+
+class GeopServiceActor extends Actor with GeopService {
+  def actorRefFactory = context
+  def receive = runRoute(myRoute)
+}
+
+trait GeopService extends HttpService {
+  import scala.util._
+
+  import MediaTypes._
+  import UnattractiveButTemporary._
+
+  val myRoute =
```

Comment:
The `spark-jobserver` route seems like it could be more maintainable in the long run (worth mentioning that I haven't spent much time with it yet). Mainly because it already has a standard interface, but can also handle job submission a few different ways:

- https://github.com/spark-jobserver/spark-jobserver#package-jar---send-to-server
- https://github.com/spark-jobserver/spark-jobserver#ad-hoc-mode---single-unrelated-jobs-transient-context
- https://github.com/spark-jobserver/spark-jobserver#persistent-context-mode---faster--required-for-related-jobs

The persistent context mode seems very similar to what we're trying to do here, but more flexible because another context could be added for different job types, or a newer version of the same job type. I also like the idea of deploying something like this vs. a long-lived `spark-submit` via Upstart.

I don't think this is something that needs to be solved in this PR, but wanted to bring it up to get an idea of where we plan on going. Being closer to the code than I am, does the persistent context mode seem like what we're trying to recreate with the Spray web service endpoints?